managers/backup-manager/BackupStatusPoller.js

```diff
@@ -0,0 +1,277 @@
+        status: {
+          'state': operationStatusResponse.state
+        }
+      }))
+      .then(() => this._logEvent(opts, operationName, operationStatusResponse))
+      .then(() => {
+        if (operationStatusResponse.operationTimedOut) {
+          const msg = `Deployment ${opts.instance_guid} ${operationName} with backup guid ${opts.backup_guid} exceeding timeout time
+        ${config.backup.backup_restore_status_poller_timeout / 1000 / 60} (mins). Stopping status check`;
+          logger.error(msg);
+        } else {
+          logger.info(`Instance ${opts.instance_guid} ${operationName} for backup guid ${opts.backup_guid} completed -`, operationStatusResponse);
+        }
+        return operationStatusResponse;
+      });
+  }
+
+  static _logEvent(opts, operation, operationStatusResponse) {
+    const eventLogger = EventLogInterceptor.getInstance(config.external.event_type, 'external');
+    const check_res_body = true;
+    const resp = {
+      statusCode: 200,
+      body: operationStatusResponse
+    };
+    if (CONST.URL[operation]) {
+      return eventLogger.publishAndAuditLogEvent(CONST.URL[operation], CONST.HTTP_METHOD.POST, opts, resp, check_res_body);
+    }
+  }
+
+  static start() {
+    function poller(object, intervalId) {
+      const resourceDetails = eventmesh.apiServerClient.parseResourceDetailsFromSelfLink(object.metadata.selfLink);
+      // If no lockedByPoller annotation then set annotation  with time
+      // Else check timestamp if more than specific time than start polling and change lockedByPoller Ip
+      return eventmesh.apiServerClient.getResource({
+          resourceGroup: resourceDetails.resourceGroup,
+          resourceType: resourceDetails.resourceType,
+          resourceId: object.metadata.name,
+        })
+        .then(resourceBody => {
+          const options = resourceBody.spec.options;
+          const response = resourceBody.status.response;
+          const pollerAnnotation = resourceBody.metadata.annotations.lockedByTaskPoller;
+          logger.debug(`Backup status pollerAnnotation is ${pollerAnnotation} current time is: ${new Date()}`);
+          return Promise.try(() => {
+            // If task is not picked by poller which has the lock on task for config.backup.backup_restore_status_check_every + BACKUP_RESOURCE_POLLER_RELAXATION_TIME then try to acquire lock
+            if (pollerAnnotation && (JSON.parse(pollerAnnotation).ip !== config.broker_ip) && (new Date() - new Date(JSON.parse(pollerAnnotation).lockTime) < (config.backup.backup_restore_status_check_every + CONST.BACKUP_RESOURCE_POLLER_RELAXATION_TIME))) { // cahnge this to 5000
+              logger.debug(`Process with ip ${JSON.parse(pollerAnnotation).ip} is already polling for task`);
+            } else {
+              const patchBody = _.cloneDeep(resourceBody);
+              const metadata = patchBody.metadata;
+              const currentAnnotations = metadata.annotations;
```

Comment:
Can we have like **_.get(patchBody, 'metadata. annotations')** as it null safe as well.

Reply:
Done